backend/controllers/userControllers.js

```diff
@@ -71,8 +75,81 @@ const authUser = asyncHandler(async (req, res) => {
     });
   } else {
     res.status(401);
-    throw new Error("Invalid Email or Password");
+    throw new Error("Invalid Password");
+  }
+});
+
+const forgotPass = asyncHandler(async (req, res) => {
+  const { email } = req.body;
+
+  try {
+    const user = await User.findOne({ email });
+    if (!user) {
+      res.status(400);
+      throw new Error("User with this email does not exist");
+    }
+    // Generate token
+    const generateOtp = () => {
+      return Math.floor(100000 + Math.random() * 900000).toString(); // 6-digit OTP
+    };
+    const resetToken = generateOtp();
+    user.resetOtp = resetToken;
+    user.resetOtpExpires = Date.now() + 3600000; // 1 hour expiry
+    await user.save();
+    
+    // console.log(`resetToken: ${resetToken}`);
+
+    const transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        user: process.env.EMAIL_USERNAME,
+        pass: process.env.EMAIL_PASSWORD,
+      },
+    });
+
+    const otp = resetToken;
+    const mailOptions = {
+      from: process.env.EMAIL_USERNAME,
```

Comment:
handle the case if there's no email_username in env

Reply:
Ok